metrics-exporter-prometheus/src/protobuf.rs

```diff
@@ -0,0 +1,327 @@
+//! Protobuf serialization support for Prometheus metrics.
+
+use indexmap::IndexMap;
+use metrics::Unit;
+use prost::Message;
+use std::collections::HashMap;
+
+use crate::common::Snapshot;
+use crate::distribution::Distribution;
+use crate::formatting::sanitize_metric_name;
+
+// Include the generated protobuf code
+mod pb {
+    #![allow(missing_docs, clippy::trivially_copy_pass_by_ref, clippy::doc_markdown)]
+    include!(concat!(env!("OUT_DIR"), "/io.prometheus.client.rs"));
+}
+
+#[cfg(feature = "http-listener")]
+pub(crate) const PROTOBUF_CONTENT_TYPE: &str =
+    "application/vnd.google.protobuf; proto=io.prometheus.client.MetricFamily; encoding=delimited";
+
+/// Renders a snapshot of metrics into protobuf format using length-delimited encoding.
+///
+/// This function takes a snapshot of metrics and converts them into the Prometheus
+/// protobuf wire format, where each `MetricFamily` message is prefixed with a varint
+/// length header.
+#[allow(clippy::too_many_lines)]
+pub(crate) fn render_protobuf(
+    snapshot: &Snapshot,
+    descriptions: &HashMap<String, (metrics::SharedString, Option<Unit>)>,
+    global_labels: &IndexMap<String, String>,
+    counter_suffix: Option<&'static str>,
+) -> Vec<u8> {
+    let mut output = Vec::new();
+
+    // Process counters
+    for (name, by_labels) in &snapshot.counters {
+        let sanitized_name = sanitize_metric_name(name);
```

Comment:
Just a _general_ note: we might need to/want to evaluate, as part of a future PR, relaxing this sanitization since I _think_ full UTF-8 metric names/tags are allowed for the Protobuf payloads? (and the text format, if you use a recent enough version and escape properly, yadda yadda)

Reply:
Presumably none of that escaping is required for the protobuf format. I'm happy to drop if you'd like? These lines were not added with a ton of thought :-/